### PR TITLE
Deadlock on supplementary nodes

### PR DESCRIPTION
**Pull request by foxware00**
I've been playing with `configureSupplementaryView` on CollectionNode and it sets `ASCollectionSectionedDataSource.configureSupplementaryViewBlockNotSet` which calls through to `DispatchQueue.main.sync` which quite frequently causes dealocks on the main thread. I noticed a discussion in #23 that applied this fix to the `configureCellBlockNotSet` method. This change simply prevents the deadlock which applying the same changes as `configureCellBlockNotSet`. I wasn't 100% sure why the `NotSet` methods are being called when I haven't configured the datasource to use block method.

### CODE REVIEW

**Review by dangthaison91 — APPROVED** [review]
LGTM

File: RxASDataSources.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
      s.name = 'RxASDataSources'
-     s.version = '0.3.4'
+     s.version = '0.4.1'
```
Comment (dangthaison91) [inline RxASDataSources.podspec:3]: Could you please revert pod version or help to combine with PR #35 for version 0.4 that supports Swift 5?

**Review by dangthaison91 — CHANGES_REQUESTED** [review]

**Review by dangthaison91 — APPROVED** [review]
Thanks @foxware00 for your supporting.

File: RxASDataSources.podspec (hunk 1 above)
Comment (foxware00) [reply RxASDataSources.podspec:3]: @dangthaison91 Done

**Comment by foxware00** [comment]
> Thanks @foxware00 for your supporting.

No problem. Are you able to make it available via cocoapods

**Comment by dangthaison91** [comment]
@foxware00 just pushed version `0.4.1`

```
 🎉  Congrats

 🚀  RxASDataSources (0.4.1) successfully published
 📅  September 2nd, 16:46
 🌎  https://cocoapods.org/pods/RxASDataSources
 👍  Tell your friends!
```

**Comment by foxware00** [comment]
@dangthaison91 Thank you for the quick turnaround!

**Pull request merged**